Userland/Applications/PixelPaint/ImageMasking.cpp

```diff
@@ -0,0 +1,181 @@
+        for (int x = 0; x < m_reference_mask->width(); x++) {
+            reference_mask_pixel = m_reference_mask->get_pixel(x, y);
+            if (!reference_mask_pixel.alpha())
+                continue;
+
+            content_pixel = m_editor->active_layer()->content_bitmap().get_pixel(x, y);
+            if (!content_pixel.alpha()) {
+                m_editor->active_layer()->mask_bitmap()->set_pixel(x, y, reference_mask_pixel.with_alpha(0));
+                continue;
+            }
+
+            current_content_luminosity = content_pixel.luminosity();
+
+            if (current_content_luminosity >= min_luminosity_start && current_content_luminosity < min_luminosity_full && has_start_range) {
+                approximation_alpha = reference_mask_pixel.alpha() * static_cast<float>((current_content_luminosity - min_luminosity_start)) / (min_luminosity_full - min_luminosity_start);
+                m_editor->active_layer()->mask_bitmap()->set_pixel(x, y, reference_mask_pixel.with_alpha(approximation_alpha));
+            } else if (current_content_luminosity > max_luminosity_full && current_content_luminosity <= max_luminosity_end && has_end_range) {
+                approximation_alpha = reference_mask_pixel.alpha() * (1 - static_cast<float>((current_content_luminosity - max_luminosity_full)) / (max_luminosity_end - max_luminosity_full));
+                m_editor->active_layer()->mask_bitmap()->set_pixel(x, y, reference_mask_pixel.with_alpha(approximation_alpha));
+            } else if (current_content_luminosity >= min_luminosity_full && current_content_luminosity <= max_luminosity_full) {
+                m_editor->active_layer()->mask_bitmap()->set_pixel(x, y, reference_mask_pixel);
+            } else {
+                m_editor->active_layer()->mask_bitmap()->set_pixel(x, y, reference_mask_pixel.with_alpha(0));
+            }
+        }
+    }
+
+    m_editor->active_layer()->did_modify_bitmap();
+    m_did_change = true;
+}
+
+ErrorOr<void> ImageMasking::ensure_reference_mask()
+{
+    if (m_reference_mask.is_null())
+        m_reference_mask = MUST(m_editor->active_layer()->mask_bitmap()->clone());
+
+    return {};
+}
+
+void ImageMasking::cleanup_resources()
+{
+    if (m_reference_mask)
+        m_reference_mask = nullptr;
+}
+
+void RangeIllustrationWidget::paint_event(GUI::PaintEvent&)
+{
+    GUI::Painter painter(*this);
+    painter.fill_rect(Gfx::IntRect(0, 0, width(), height()), palette().color(background_role()));
+    float fraction = width() / 255.0f;
+
+    Gfx::Path illustration;
+    illustration.move_to(Gfx::FloatPoint(fraction * m_edge_mask_values->lower_range(), height()));
+    illustration.line_to(Gfx::FloatPoint(fraction * m_full_mask_values->lower_range(), 0));
+    illustration.line_to(Gfx::FloatPoint(fraction * m_full_mask_values->upper_range(), 0));
+    illustration.line_to(Gfx::FloatPoint(fraction * m_edge_mask_values->upper_range(), height()));
```

Comment:
Instead of repeating `Gfx::FloatPoint` 4 times, you can just use `{ fraction * ..., height() }` etc. here